mech_client/interact.py

```diff
@@ -178,39 +221,64 @@ def watch_for_events(
         no_receipt = True
         while no_receipt:
             try:
-                tx_receipt = ethereum_ledger_api._api.eth.get_transaction_receipt(
-                    tx_hash
-                )
+                tx_receipt = ledger_api._api.eth.get_transaction_receipt(tx_hash)
                 no_receipt = False
             except Exception:
                 time.sleep(1)
         event_signature = tx_receipt["logs"][0]["topics"][0].hex()
         if event_signature == EVENT_SIGNATURE_REQUEST:
-            rich_logs = contract_instance.events.Request().processReceipt(tx_receipt)
+            rich_logs = mech_contract.events.Request().processReceipt(tx_receipt)
             request_id = rich_logs[0]["args"]["requestId"]
             print(f"Request on-chain with id: {request_id}")
         if event_signature == EVENT_SIGNATURE_DELIVER:
-            rich_logs = contract_instance.events.Deliver().processReceipt(tx_receipt)
+            rich_logs = mech_contract.events.Deliver().processReceipt(tx_receipt)
             data = rich_logs[0]["args"]["data"]
             request_id_ = rich_logs[0]["args"]["requestId"]
             if request_id != request_id_:
                 continue
             data_url = "https://gateway.autonolas.tech/ipfs/f01701220" + data.hex()
             print(f"Data arrived: {data_url}")
             is_waiting = False
+
     response = requests.get(data_url + "/" + str(request_id))
     response_json = response.json()
     result = response_json["result"]
     print(f"Data:\n{result}")
     return response_json
 
 
-def interact(prompt: str, tool: str) -> Dict[str, Any]:
-    ethereum_crypto = EthereumCrypto(private_key_path=PRIVATE_KEY_FILE_PATH)
-    ethereum_ledger_api = EthereumApi(**ETHEREUM_TESTNET_CONFIG)
-    wss = register_event_handlers(ethereum_crypto)
-    contract_instance = send_request(ethereum_crypto, ethereum_ledger_api, prompt, tool)
-    response = watch_for_events(
-        wss, contract_instance, ethereum_ledger_api, ethereum_crypto
+def interact(
+    prompt: str,
+    agent_id: int,
+    tool: Optional[str] = None,
```

Comment:
tool is not optional?

also keep the order so that we don't unnecessarily make it harder to upgrade

Reply:
Tools IS optional here, if the user already know which tool they want to use they can provide it using `--tool` flag on cli and if they don't provide the tool name they'll be prompted with available tools to select from. Please read the updated `README.md`